src/targets/maven.ts

```diff
@@ -0,0 +1,250 @@
+  settingsPath = 'MAVEN_SETTINGS_PATH',
+  mavenRepoUrl = 'MAVEN_REPO_URL',
+  mavenRepoId = 'MAVEN_REPO_ID',
+  mavenCliPath = 'MAVEN_CLI_PATH',
+  gradleCliPath = 'GRADLE_CLI_PATH',
+}
+
+// Paths should be relative to the root of the repository
+const DEFAULT_ENV_VARIABLES: Record<string, string> = {
+  MAVEN_DISTRIBUTIONS_PATH: 'distributions/',
+  MAVEN_SETTINGS_PATH: 'scripts/settings.xml',
+  MAVEN_REPO_URL:
+    'https://oss.sonatype.org/service/local/staging/deploy/maven2/',
+  MAVEN_REPO_ID: 'ossrh',
+  MAVEN_CLI_PATH: 'scripts/mvnw.cmd',
+  GRADLE_CLI_PATH: 'gradlew',
+};
+
+/**
+ * Target responsible for uploading files to Maven Central.
+ */
+export class MavenTarget extends BaseTarget {
+  /** Target name */
+  public readonly name: string = 'maven';
+  /** Target options */
+  public readonly mavenConfig: MavenTargetConfig;
+
+  public constructor(
+    config: TargetConfig,
+    artifactProvider: BaseArtifactProvider
+  ) {
+    super(config, artifactProvider);
+    this.mavenConfig = this.getMavenConfig();
+    this.checkRequiredSoftware();
+  }
+
+  private getMavenConfig(): MavenTargetConfig {
+    return {
+      // Required env variables
+      ossrhUsername: this.getEnvVarValue(RequiredConfig.ossrhUsername),
+      ossrhPassword: this.getEnvVarValue(RequiredConfig.mavenPassword),
+      // TODO: MAVEN_CENTRAL_* shouldnt be required if the user already has `gradle.properties`
+      mavenUsername: this.getEnvVarValue(RequiredConfig.mavenUsername),
+      mavenPassword: this.getEnvVarValue(RequiredConfig.mavenPassword),
+      // Optional env variables
+      distributionsPath: this.getEnvVarOrDefault(
+        OptionalConfig.distributionsPath
+      ),
+      settingsPath: this.getEnvVarOrDefault(OptionalConfig.settingsPath),
+      mavenRepoUrl: this.getEnvVarOrDefault(OptionalConfig.mavenRepoUrl),
+      mavenRepoId: this.getEnvVarOrDefault(OptionalConfig.mavenRepoId),
+      mavenCliPath: this.getEnvVarOrDefault(OptionalConfig.mavenCliPath),
+      gradleCliPath: this.getEnvVarOrDefault(OptionalConfig.gradleCliPath),
+    };
+  }
+
+  private getEnvVarValue(envVar: RequiredConfig): string {
+    if (process.env[envVar]) {
+      return process.env[envVar] as string; // `as string` to make TS happy
+    }
+    throw new ConfigurationError(
+      `Cannot publish to Maven Central: missing credentials.
+      Please, use the ${envVar} environment variable.`
+    );
+  }
+
+  private getEnvVarOrDefault(envVar: OptionalConfig): string {
+    if (process.env[envVar]) {
+      return process.env[envVar] as string; // `as string` to make TS happy
+    }
+    return DEFAULT_ENV_VARIABLES[envVar];
+  }
+
+  /**
+   * Checks whether the required software to run this target is available
+   * in the system. It assumes the config for this target to be available.
+   */
+  private checkRequiredSoftware(): void {
+    this.logger.debug('Checking Maven CLI is available...');
+    checkExecutableIsPresent(this.mavenConfig.mavenCliPath);
```

Comment:
Adding the paths that are being checked to the debug log would make them way more useful.